### ISSUE TITLE
Create own Color class instead of using java.awt.Color

### ISSUE THREAD

**Issue by CrendKing** [robvanderleek/JLifx#91]
Since java.awt cannot be used in Android, this one Color class would prevent the library's color management function to be used in Android. Suggest to replace it with your own Color class, a relatively small effort to greatly improve portability.

**Comment by robvanderleek** [comment]
Hi @CrendKing 😄 
As you might have noticed I'm not actively working on this repo. Your suggestion seems valid (I'm not developing for Android myself).
Do you have a use-case for jlifx? If so I can give this issue some priority.
Rob

**Comment by CrendKing** [comment]
Well. I have a LIFX bulb myself. I wrote a small android widget myself that turns on/off and (planned to) control the color/brightness the bulb directly on the Android launcher. This is easier and faster to use than the official LIFX app in the situation, because the LAN protocol can send request directly to the bulb (provided you know the MAC and IP) instead connecting to the LIFX cloud first.

Here is a Python version of what I just mentioned above, using https://github.com/mclarkk/lifxlan:
```python
from lifxlan import Light
import sys


def power_on(light):
    light.set_colortemp(4000)
    light.set_brightness(65535)
    light.set_power(65535)


def power_off(light):
    light.set_power(0)


if __name__ == '__main__' and len(sys.argv) >= 2:
    living = Light(<MAC>, <IP>)

    command = sys.argv[1]
    if command == 'on':
        power_on(living)
    elif command == 'off':
        power_off(living)
    elif command == 'toggle':
        if living.get_power() == 0:
            power_on(living)
        else:
            power_off(living)
```
Unfortunately, JLifx does not seem to have high level interface that send request directly to bulb. I had to use `GatewayBulb` and construct `StatusResponsePacket` directly. Something like this:
```java
final InetAddress bulbIp = InetAddress.getByName(<IP>);
final byte[] bulbMacBytes = { <MAC> };
final MacAddress bulbMac = new MacAddress(bulbMacBytes);
_bulb = new GatewayBulb(bulbIp, bulbMac);

final List<StatusResponsePacket> packets = _packetService.sendStatusRequestPacket(_bulb);
for (StatusResponsePacket packet : packets) {
    if (packet.getType() == 0x6B) {
        _packetService.sendPowerManagementPacket(_bulb, packet.getPower() == 0);
        // cannot use sendColorManagementPacket() here
        break;
    }
}
```
Hope this explains the issue. Thanks.

**Comment by robvanderleek** [comment]
Hi @CrendKing 
I've refactored JLifx a bit over the weekend.  All uses of `java.awt` are removed. Version 0.10 also offers a high-level bulb interface for power and color operations, for an example see: https://github.com/robvanderleek/JLifx/blob/master/src/test/java/io/github/robvanderleek/jlifx/examples/ColorBulbByIpAddress.java
Let me know if this helps with your use-case.

**Comment by CrendKing** [comment]
Thank for the work. The change of `Color` should work for my use case. Unfortunately, I think you might have misunderstood my requirement of the bulb addressing. What I want is to completely skip the "discover" step and directly send power on/off request to the bulb, because I can statically assign each bulb an IP in my router and hard code that IP in my app.

Imagine I have 200 bulbs in my home. I specifically know that bulb "foo" is at IP 192.168.0.54 and its Mac address is BAR. Instead of broadcasting discovery packets to the LAN and waiting for all the 200 responses then filter by IP, I want to `Bulb b = new Bulb("192.168.0.54", "BAR"); b.switchOn();` Done. This way it should be way faster if the user is only making changes to subset of the local bulbs.

**Comment by robvanderleek** [comment]
Currently you are steering the direction of this project and so far your suggestions look like improvements to me 😄 

I've basically implemented the interface you propose (`status` is updated by the high-level control methods in `Bulb`, this makes a lot of sense). See https://github.com/robvanderleek/JLifx/blob/master/src/test/java/io/github/robvanderleek/jlifx/examples/ConnectWithIpAddressAndMacAddress.java for an example of how to directly connect to a Bulb, given you know the IP and MAC address.

I've tagged 0.11 and it has been released to Maven Central, should be visible in a couple of minutes.

If you have more suggestions or want to open a PR for stuff let me know. Making the `kelvin` value configurable is something for a separate issue but I don't have time to work on that now.

Good luck with your widget!
